phaser/steps.py

```diff
@@ -34,7 +35,13 @@ def _batch_step_wrapper(batch, context=None, __probe__=None):
         if __probe__ == PROBE_VALUE:
             return BATCH_STEP
         try:
-            result = step_function(batch, context=context)
+            # LMDTODO: Discovered in testing that if the step function doesn't declare the context in its params,
+            # when the calling code tries to pass it a batch AND a context, it doesn't even run and its hard to see why.
+            # See test test_batch_step_missing_param. The TODO is to fix this for the other step types as well.
+            if 'context' in str(inspect.signature(step_function)):
```

Comment:
This is sufficient for now, but relies on the user having named the parameter `context`.  Once we put in typing, then we should change this to check that the type is `Context`, if that is possible.

Reply:
Then again, relying on the stringification of `signature` is a little hacky.  Would be better as `if 'context' in inspect.signature(step_function).parameters.keys()`.

Reply:
thank you, good find.